src/state.rs

```diff
@@ -0,0 +1,56 @@
+//! This module is dedicated for the state dump of bspwm. It can be received by
+//! running `bspc wm -d` your shell.
+
+use crate::errors::ReplyError;
+use crate::socket::{connect, BspcCommunication};
+use crate::tree::Monitor;
+use crate::Id;
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct FocusHistoryEntry {
+    pub monitor_id: Id,
+    pub desktop_id: Id,
+    pub node_id: Id,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct State {
+    pub focused_monitor_id: Id,
+    pub clients_count: u32,
+    pub monitors: Vec<Monitor>,
+    pub focus_history: Vec<FocusHistoryEntry>,
+    pub stacking_list: Vec<Id>,
+}
+
+/// Returns a dump of the current bspwm state <br>
+/// Contains all monitors with their desktops (and window respectively)
+pub fn get_current_state() -> Result<State, ReplyError> {
+    let mut conn = connect()?;
```

Comment:
I would stick to the style `socket::connect()?` here. Also, you don't need `format` macro here. So, I would like it to be changed to
```rust
let mut conn = socket::connect()?;